tools/serverpod_cli/lib/src/generator/dart/library_generators/server_test_tools_generator.dart

```diff
@@ -565,6 +568,12 @@ extension on ParameterDefinition {
           ]),
         ]).code,
       ]);
+    } else if (isModelFromProject) {
+      if (type.nullable) {
```

Comment:
A bit surprised that I had to add this, but otherwise when using a plain model class as an argument to an endpoint method, the test would crash, because the model (not it's JSON representation) would be given to `Protocol.deserialize`.

Reply:
> How does this work with models that are part of a module?

I'll add a test for that. Will be interesting how I can identify those, as the problem here was that no part of the stack seems to call `toJson` on the model class, but on the receiving end it is passed to `deserialize` which expect the JSON "map" representation (and not a class instance, which would already be the result).

This was the best I could come up with to detect when it would be allowed (and necessary) to call `toJson()` on the parameter.

> Or `Custom` models?

What do you mean with "custom" here? It currently does work fine (and is tested) for models defined in the same project.

Reply:
Changed this to generate "toJSON" (though we have to use `SerializationManager.encode` (which gives a `String`) + `jsonDecode` in order to get the JSON "Map" representation, as the model's `toJson` is not "deep").

Now this also works for external models.